### ISSUE TITLE
Logging In

### ISSUE THREAD

**Issue by scouseman** [zoffline/zwift-offline#35]
I am also having a problem logging into zwift profile.
I have ran the get_profile script successfully and it has created the profile.bin file.
I have put the profile.bin file into the storage folder but when I run zwift with zoffline, I can't get passed the zwift login page, error ' Failed to login check internet connection'

**Comment by oldnapalm** [comment]
Your issue does not seem to be related to your profile.

Does it work without profile.bin in the storage folder?

Did you go through all installation steps, including client configuration?

If you have installed (or updated to) current Zwift version (1.0.46903) you need to update zwift-offline\cdn\gameassets\Zwift_Updates_Root\Zwift_ver_cur.xml (just copy that file from Zwift folder).

**Comment by scouseman** [comment]
Thanks for your quick reply, yes I am currently using the latest version of zwift (1.0.46903) and I  copied the Zwift_ver_cur.xml file over.
Without Profile.bin I still get stuck at the login page.
There are also no errors on the server

**Comment by oldnapalm** [comment]
Without profile.bin you should be prompted to create a new profile.

Did you import cert-zwift-com.p12 and append cert-zwift-com.pem to cacert.pem?

Make sure you allow python.exe in Windows firewall.

**Comment by scouseman** [comment]
Yes I am able to get to the create profile page but when I fill the detail in it won't proceed any further.
I will re-do the cert and check the firewall
Thanks again for your time

**Comment by scouseman** [comment]
It appears it was a firewall issue, all resolved now.
Thanks very much.

**Comment by scouseman** [comment]
I just want to thank, you guys for the work you are doing here, it is very much appreciated.
This is exactly what I have been looking for.I plan to use zoffline for doing my custom workouts which I got from Trainerday (a great free site). I don't know how viable it is but a great feature if it was possible would be if you could race against your ghost, previous best time on the same workout,/course.
But anyway thanks again.

**Comment by oldnapalm** [comment]
Earlier versions of the ZwiftMap app had the ghost feature, but it was removed because Zwift requested developers to stop using its API.

https://gitlab.com/jeroni7100/zwiftmap-multi/-/tree/69fbc89663cef8af278cf43e54daef3b7bbd8e09

I tried to compile that version using zwift-mobile-api 0.3.9 and zwift-second-screen 0.1.1 (versions from before the developer API changes) but it didn't work with zoffline. I guess it's because the UDP game node is not implemented.

Missing UDP node is also the cause of issue #1 then I believe we won't see ghosts before that issue is resolved.

It would be a great feature indeed.
